### ISSUE TITLE
Exception in return_extracellular_spike when generating templates

### ISSUE THREAD

**Issue by cristi-zz** [alejoe91/MEArec#23]
Hi!
I am running ```mearec gen-templates -p SqMEA-10-15um``` and I get these errors:

```
 L5_BTC_bAC217_1 10 / 13 


NEURON -- VERSION 7.5 master (6b4c19f) 2017-09-25
Duke, Yale, and the BlueBrain Project -- Copyright 1984-2016
See http://neuron.yale.edu/neuron/credits

/home/cristi/miniconda3/envs/py6/lib/python3.6/site-packages/LFPy/cell.py:36: UserWarning: LFPy could not read NEURON version info. v7.6.4 or newer required
  warn('LFPy could not read NEURON version info. v7.6.4 or newer required')
Simulating  L5_BTC_bAC217_1
loading membrane mechanisms from ../mods/x86_64/.libs/libnrnmech.so
Additional mechanisms from files
 CaDynamics_E2.mod Ca_HVA.mod Ca_LVAst.mod Ca.mod Ih.mod Im.mod K_Pst.mod K_Tst.mod Nap_Et2.mod NaTa_t.mod NaTs2_t.mod ProbAMPANMDA_EMS.mod ProbGABAAB_EMS.mod SK_E2.mod SKv3_1.mod
Setting temperature to 34.000000 C
Setting simulation time step to 0.025000 ms
	1 
	1 
	1 
template bAC217_L5_BTC_160272643a exist already
Input weight:  0.25  - Num Spikes:  15
Simulating  L5_BTC_bAC217_1
Mechanisms already loaded from path: ../mods.  Aborting.
template bAC217_L5_BTC_160272643a exist already
Cell  L5_BTC_bAC217_1  extracellular spikes to be simulated
Traceback (most recent call last):
  File "/home/cristi/miniconda3/envs/py6/lib/python3.6/site-packages/MEArec/simulate_cells.py", line 854, in <module>
    calc_extracellular(cell_model, extra_sim_folder, vm_im_sim_folder, **params)
  File "/home/cristi/miniconda3/envs/py6/lib/python3.6/site-packages/MEArec/simulate_cells.py", line 462, in calc_extracellular
    [x_lim, y_lim, z_lim], rotation, pos=position)
  File "/home/cristi/miniconda3/envs/py6/lib/python3.6/site-packages/MEArec/simulate_cells.py", line 795, in return_extracellular_spike
    x_rand = np.random.uniform(limits[0][0], limits[0][1])
IndexError: tuple index out of range

```

While I get an output message, that the templates were saved in a file, that HDF5 file has only 6KB (so empty i guess)

My python (conda)  env is:
```
python                    3.6.7  
neuron                    7.5  
meautility                1.2.2
lfpy                      2.0.1 
mearec                    1.1.3 
```

Everything was freshly installed, using conda and pip. 

If it helps, when I am running the template generator from python I got a different error (for each thread):

```
Traceback (most recent call last):
  File "/home/cristi/miniconda3/envs/py6/lib/python3.6/site-packages/MEArec/simulate_cells.py", line 17, in <module>
    import MEAutility as MEA
ImportError: No module named MEAutility
Exiting Thread-4

```

Of course, running a MEAutility command, few lines above the ```mr.gen_templates()```  works (eg ```MEA.return_mea(probe_name)```. And the template generator code follows the documentation. 

Thank you!

**Comment by alejoe91** [comment]
Hi @cristi-zz 

The problem was in passing the xlim, ylim, and zlim args from the cli command. Now it checks if the length of these args is 2.

Regarding the second problem, I was not able to reproduce it.
I tried:
```
import MEArec as mr
tempgen = mr.gen_templates('path-to-cell-models')
```
and it runs fine. Are you sure you are running the right ipython? Sometimes in anaconda you need to install ipython in the environment.

Fixed in the latest commit 548d31a3cf1c9c3b5a6c09c8bcff160bde6f3a21. I'm uppdating pypi now. Can you test that the cli works?

Cheers
Alessio

**Comment by cristi-zz** [comment]
Hi! Thank you!

Except the cell folder and output folder, no other options were touched.
I did pip --upgrade to MEArec 1.1.4 and re-run the templates instruction (cli)
I got an 18MB HDF5 and no visible errors. For the 2nd problem I will post an independent issue if it will block me. But command line is enough for now.

I will move on to generating the recordings soon, but, for now, thank you for the fix!
